### ISSUE TITLE
NOT SUCCESSFULLY CONNECTED?

### ISSUE THREAD

**Issue by lazarte** [socketio/socket.io-client-swift#594]
Hi, I successfully setup the SocketIOClient and this seems to be that that the server replied when I connect. But the client doesn't receive response that it was connected. When trying to emit will give me the error that it was not connected. See below codes and logs. Please help.

`    func connection() {
        let url = URL(string: "SOCKET_URL")!
        socket = SocketIOClient(socketURL: url,
                                     config: [.log(true),
                                              .forcePolling(true),
                                              .path("/primus"),
                                              .forceWebsockets(true)
            ])
        
        socket.onAny {print("Any Got event: \($0.event), with items: \($0.items)")}
        
        socket.on("connect") {data, ack in
            print("socket connected")
        }
        
        socket.connect()
    }`

LOG SocketIOClient: Adding handler for event: connect
LOG SocketIOClient: Adding handler for event: connected
LOG SocketIOClient: Adding handler for event: message
LOG SocketIOClient: Adding handler for event: data
LOG SocketIOClient: Adding engine
LOG SocketEngine: Starting engine. Server: SOCKET_URL
LOG SocketEngine: Handshaking
LOG SocketEngine: Got message: {"type":0,"data":["@HandShake","Hello"]}
LOG SocketIOClient: Handling event: error with data: ["Tried emitting data when not connected"]
Any Got event: error, with items: Optional(["Tried emitting data when not connected"])

**Comment by nuclearace** [comment]
That's a pretty straightforward error. You shouldn't be trying to send something before the connect event is fired. But are you actually connecting to a socket.io server?

**Comment by lazarte** [comment]
Yes I'm connecting to the primus with socket.io transformer. I got an message from the server so meaning the handshake is successful?

**Comment by nuclearace** [comment]
That got message doesn't look like something socket.io would send. The first thing socket.io should be sending is a connect packet that has some connection information.

**Comment by lazarte** [comment]
Actually that message emit from our server. Manually added this code from this pull request [489](https://github.com/socketio/socket.io-client-swift/pull/489) seems to trigger the connect event?

**Comment by nuclearace** [comment]
Are you sure that you're not looking for a straight Websocket client? Because if you're changing to so much that it isn't getting the proper packets to connect you're getting into fork land.

**Comment by lazarte** [comment]
@nuclearace below is the code for the primus server don't know why the client doesn't trigger the connect event. On the other hand using socket.io works fine. But want to know why its not working with the primus.

var Primus = require('primus')
var primus = Primus.createServer({
  port: 3002, transformer: 'websockets'
});
primus.on('connection', function (spark) {
  spark.write({"type":0,"data":["@HandShake","Hello"]});
});

**Comment by nuclearace** [comment]
Does `primus` follow the socket.io protocol? Because that's the only way this client would work with it.

**Comment by khodzhamberdiyev** [comment]
Same problem. Absent connect event. Xcode 8.2.1 and Swift 3.0.2. Log:
```
LOG SocketIOClient: Adding handler for event: connect
LOG SocketIOClient: Adding handler for event: currentAmount
LOG SocketIOClient: Adding engine
LOG SocketEngine: Starting engine. Server: ws://xxx.xxx.xxx.xxx:8080
LOG SocketEngine: Handshaking
```
